### PR TITLE
don't re-use connection objects on windows and mac either

### PR DESCRIPTION
**Pull request by dothebart**
This fixes `ssl_server` cluster tests failing on mac & windows.

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/Mac/job/arangodb-ANY-mac-matrix/580/
http://jenkins01.arangodb.biz:8080/view/Windows/job/arangodb-ANY-windows-matrix/760/

File: 3rdParty/fuerte/src/GeneralConnection.cpp [hunk 1]
```diff
@@ -75,12 +75,10 @@ void GeneralConnection<ST>::shutdownConnection(const Error err,
   auto state = _state.load();
   if (state != Connection::State::Failed) {
     state = Connection::State::Disconnected;
-#ifdef __linux__
-    // hack to fix SSL streams on linux
+    // hack to fix SSL streams
     if (_config._socketType == SocketType::Ssl) {
```
Comment (graetzer) [inline 3rdParty/fuerte/src/GeneralConnection.cpp]:
```suggestion
    if constexpr (ST == SocketType::Ssl) {
```

**Comment by dothebart** [comment]
tests mostly green.

**Comment by KVS85** [comment]
Don't need backporting since no failures like this were observer for 3.5- branches (which doesn't use Fuerte here).

**Pull request merged**